### PR TITLE
kubelet: Make cadvisor GetContainerInfo tests table driven

### PR DESCRIPTION
**Pull request by nhlfr**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34931)

### CODE REVIEW

**Comment by vishh** [comment]
@dashpole can you review this?

**Comment by dashpole** [comment]
ill take a look

**Comment by dashpole** [comment]
*[pkg/kubelet/kubelet_cadvisor_test.go, line 190 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931#-KVXcgwvn_RTmUy0I6sa:-KVXcgwvn_RTmUy0I6sb:b-41h1i3) ([raw file](https://github.com/kubernetes/kubernetes/blob/0be3f144efd0df8d789ac9a1e404fd180ce2ef48/pkg/kubelet/kubelet_cadvisor_test.go#L190)):*
> ```Go
> 		stats, err := kubelet.GetContainerInfo(tc.requestedPodFullName, tc.requestedPodUid, tc.requestedContainerName, cadvisorReq)
> 		if err != tc.expectedError {
> 			t.Errorf("expected error %#v, got %#v", tc.expectedError, err)
> ```

nit: give each test a name field, similar to what old test names were.  Include the names in any errors, so we know which of the tests in the list failed.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931)*

**Comment by dashpole** [comment]
*[pkg/kubelet/kubelet_cadvisor_test.go, line 193 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931#-KVXdCTGYyVzWF6CGOax:-KVXdCTGYyVzWF6CGOay:b51l13j) ([raw file](https://github.com/kubernetes/kubernetes/blob/0be3f144efd0df8d789ac9a1e404fd180ce2ef48/pkg/kubelet/kubelet_cadvisor_test.go#L193)):*
> ```Go
> 		}
> 		if tc.expectStats && stats == nil {
> 			t.Fatalf("stats should not be nil")
> ```

nit: same as above

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931)*

**Comment by dashpole** [comment]
LGTM after nits are fixed

**Comment by nhlfr** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/kubelet/kubelet_cadvisor_test.go, line 190 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931#-KVXcgwvn_RTmUy0I6sa:-KVcuB1wfDsCpcG5iJTV:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/0be3f144efd0df8d789ac9a1e404fd180ce2ef48/pkg/kubelet/kubelet_cadvisor_test.go#L190)):*
<details><summary><i>Previously, dashpole (David Ashpole) wrote…</i></summary>
> nit: give each test a name field, similar to what old test names were.  Include the names in any errors, so we know which of the tests in the list failed.
</details>
Done.

---

*[pkg/kubelet/kubelet_cadvisor_test.go, line 193 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931#-KVXdCTGYyVzWF6CGOax:-KVcuC3qUU9kzj5XNz8L:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/0be3f144efd0df8d789ac9a1e404fd180ce2ef48/pkg/kubelet/kubelet_cadvisor_test.go#L193)):*
<details><summary><i>Previously, dashpole (David Ashpole) wrote…</i></summary>
> nit: same as above
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/34931)*

**Comment by dashpole** [comment]
LGTM, but someone else will have to apply the label

File: pkg/kubelet/kubelet_cadvisor_test.go [hunk 1]
```diff
@@ -24,44 +24,183 @@ import (
...
+			requestedPodUid:           "uuid",
+			requestedContainerName:    "foo",
+			expectDockerContainerCall: true,
+			mockError:                 cadvisorApiFailure,
+			expectedError:             cadvisorApiFailure,
+			expectStats:               false,
+		},
+		{
+			name:                      "get container info on non-existent container",
+			containerID:               "",
```
Comment (timstclair) [inline pkg/kubelet/kubelet_cadvisor_test.go]: optional: you don't need to specify default values (`""`, `false`, `nil`), so they can be omitted to shorten this. If you think it's more readable with all fields specified it's ok to leave them.

**Comment by timstclair** [comment]
needs a rebase

File: pkg/kubelet/kubelet_cadvisor_test.go (hunk 1 above)
Comment (nhlfr) [inline pkg/kubelet/kubelet_cadvisor_test.go]: That's right, I think that leaving them makes it more readable.

**Comment by nhlfr** [comment]
@timstclair done

**Pull request merged**